### ISSUE TITLE
siriproxy gencerts failing | cannot load such file -- siriproxy/command_line

### ISSUE THREAD

**Issue by ChaseFlorell** [plamoni/SiriProxy#366]
I've looked through the issues, and have found similar issues, but I haven't found any definitive resolutions.

I'm trying to get SiriProxy installed on OS X 10.8.2, and I can get all the way to the `siriproxy gencerts` without issue, but this last command is failing with the following output

> gollum:siriproxy media$ siriproxy gencerts
> /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- siriproxy/command_line (LoadError)
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in`require'
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/bin/siriproxy:4:in `<main>'
> gollum:siriproxy media$ 

What does it take to generate the certificates?

**Comment by plamoni** [comment]
Looks like it's having trouble locating the siriproxy command line tools. Might check to make sure your `rake install` went well. Also, this error log looks incomplete. Can you confirm this is everything?

**Comment by ChaseFlorell** [comment]
Here's what I get when I run `rvmsudo gem install rake bundler`

> gollum:siriproxy media$ rvmsudo gem install rake bundler
> /Users/media/.rvm/rubies/ruby-1.9.3-p194/bin/gem:4: warning: Insecure world writable dir /Users/media/.rvm/gems/ruby-1.9.3-p194@SiriProxy/bin in PATH, mode 040777
> Successfully installed rake-0.9.2.2
> Successfully installed bundler-1.2.1
> 2 gems installed
> Installing ri documentation for rake-0.9.2.2...
> Installing ri documentation for bundler-1.2.1...
> Installing RDoc documentation for rake-0.9.2.2...
> Installing RDoc documentation for bundler-1.2.1...
> gollum:siriproxy media$ 

And here's what I get when I run `rake install`

> gollum:siriproxy media$ rake install
> /Users/media/SiriProxy/siriproxy.gemspec:16: warning: Insecure world writable dir /Users/media/.rvm/gems/ruby-1.9.3-p194@SiriProxy/bin in PATH, mode 040777
> siriproxy 0.3.0 built to pkg/siriproxy-0.3.0.gem
> siriproxy (0.3.0) installed
> gollum:siriproxy media$ 

Then `siriproxy gencerts` spits out

> gollum:siriproxy media$ siriproxy gencerts
> /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- siriproxy/command_line (LoadError)
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in`require'
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/bin/siriproxy:4:in `<main>'
> gollum:siriproxy media$

**Comment by plamoni** [comment]
Hmm, I'm not sure. I'm not really a Ruby expert (believe it or not). 

One thing that looks odd is this line:

```
from /Users/media/.rvm/rubies/ruby-1.9.3-p194/bin/siriproxy:4:in 
```

On my installation, the siriproxy "binary" is located in `~/.rvm/gems/ruby-1.9.3-p194/bin" (note it's "gems" and not "rubies"). Not sure if that matters. It seems like there's maybe some sort of PATH issue?

I hate to make you start over from the beginning, but that might be worth a try... Go ahead and reinstall Ruby 1.9.3 also, if you're not using it for anything else.

**Comment by ChaseFlorell** [comment]
I'm not using ruby for anything but this. I'll try reinstalling ruby and see how it goes. 

Here's my latest attempt, I was able to get rid of the `insecure world writable dir` with this.

> gollum:siriproxy media$ sudo chmod -R 775 /Users/media/.rvm/
> gollum:siriproxy media$ rvmsudo gem install rake bundler
> Successfully installed rake-0.9.2.2
> Successfully installed bundler-1.2.1
> 2 gems installed
> Installing ri documentation for rake-0.9.2.2...
> Installing ri documentation for bundler-1.2.1...
> Installing RDoc documentation for rake-0.9.2.2...
> Installing RDoc documentation for bundler-1.2.1...
> gollum:siriproxy media$ rake install
> siriproxy 0.3.0 built to pkg/siriproxy-0.3.0.gem
> siriproxy (0.3.0) installed
> gollum:siriproxy media$ siriproxy gencerts
> /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- siriproxy/command_line (LoadError)
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in`require'
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/bin/siriproxy:4:in `<main>'
> gollum:siriproxy media$

**Comment by ChaseFlorell** [comment]
I did a reinstall as described. I can get all the way to `siriproxy gencerts` without an error now, but I still get the same error.

> gollum:~ media$ siriproxy gencerts
> /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- siriproxy/command_line (LoadError)
>     from /Users/media/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/site_ruby/1.9.1/rubygems/custom_require.rb:36:in`require'
>     from /Users/media/.rvm/gems/ruby-1.9.3-p194/bin/siriproxy:4:in `<main>'
> gollum:~ media$ 

It's also worth noting that I had to follow BenjaminWood's comments from [issue 98](https://github.com/plamoni/SiriProxy/issues/98) in order to get `siriproxy gencerts` to even run

> The gem was populated to /pkg but since I couldn't get it to install I manually moved the entire contents of the siriproxy folder to "Users/benjamin/.rvm/gems/ruby-1.9.3-p0/gems/siriproxy-0.0.1". I also had to move the "siriproxy" (no extension) file form /bin into place at "/Users/benjamin/.rvm/gems/ruby-1.9.3-p0/bin".

**Comment by plamoni** [comment]
I would guess the stuff from BenjaminWood's comments is what's causing the problem. I've installed SiriProxy under several environments (OSX and a couple flavors of Linux) and I haven't had any issues where I needed to manually relocate files like that. And the fact that you manually moved some files and now you're getting a "cannot load such file" error doesn't really feel like a coincidence to me.

If your stuff is getting put in `/bin` then there's something wrong. It should be getting placed in `~/.rvm/gems/...` automatically. I would make sure that the version of `rake` you're using is the right one. Do a `which rake` and make sure that it's showing the proper location for rake (that would be somewhere in your `~/.rvm/` folder --- for me it's: `/home/siriproxy/.rvm/gems/ruby-1.9.3-p194@global/bin/rake`). Also, make sure you're not doing `sudo rake install` or `rvmsudo rake install` that will cause issues.
